src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js

```diff
@@ -366,34 +374,88 @@ export const generateMessageBubble = (
           );
         } else {
         */
-          listItems.push(
-            generateAnswerBubble(
-              actionType,
-              index,
-              replacedText,
-              message,
-              latestUserMsgID,
-              showFeedback,
-            ),
-          );
-        //}
-        break;
-      }
-      case 'anchor': {
-        const { link, text } = action;
         listItems.push(
-          generateAnchorBubble(
+          generateAnswerBubble(
             actionType,
             index,
-            text,
-            link,
+            replacedText,
             message,
             latestUserMsgID,
             showFeedback,
           ),
         );
+        // }
         break;
       }
+      case 'anchor':
+        {
+          const { link, text } = action;
+          if (checkMapAction(link)) {
+            let mapAnchor = null;
```

Comment:
Move this entire logic of getting the lat long and other calculation to the other function defined above.